frontend/appflowy_flutter/lib/workspace/application/settings/shortcuts/settings_shortcuts_cubit.dart

```diff
@@ -0,0 +1,131 @@
+
+part 'settings_shortcuts_cubit.freezed.dart';
+
+String kCouldNotLoadErrorMsg = "Could not load shortcuts,Try again";
+String kCouldNotSaveErrorMsg = "Could not save shortcut, Try again";
+
+@freezed
+class ShortcutsState with _$ShortcutsState {
+  const factory ShortcutsState({
+    @Default(<CommandShortcutEvent>[])
+    List<CommandShortcutEvent> commandShortcutEvents,
+    @Default(ShortcutsStatus.initial) ShortcutsStatus status,
+    @Default('') String error,
+  }) = _ShortcutsState;
+}
+
+enum ShortcutsStatus { initial, updating, success, failure }
+
+class ShortcutsCubit extends Cubit<ShortcutsState> {
+  ShortcutsCubit(this.service) : super(const ShortcutsState());
+
+  final SettingsShortcutService service;
+
+  Future<void> fetchShortcuts() async {
+    emit(
+      state.copyWith(
+        status: ShortcutsStatus.updating,
+        error: '',
+      ),
+    );
+    try {
+      final customizeShortcuts = await service.getCustomizeShortcuts();
+      await service.updateCommandShortcuts(
+        commandShortcutEvents,
+        customizeShortcuts,
+      );
+      //sort the shortcuts
+      commandShortcutEvents.sort((a, b) => a.key.compareTo(b.key));
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.success,
+          commandShortcutEvents: commandShortcutEvents,
+          error: '',
+        ),
+      );
+    } catch (e) {
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.failure,
+          error: kCouldNotLoadErrorMsg,
+        ),
+      );
+    }
+  }
+
+  Future<void> updateAllShortcuts() async {
+    emit(
+      state.copyWith(
+        status: ShortcutsStatus.updating,
+        error: '',
+      ),
+    );
+    try {
+      await service.saveAllShortcuts(state.commandShortcutEvents);
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.success,
+          error: '',
+        ),
+      );
+    } catch (e) {
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.failure,
+          error: kCouldNotSaveErrorMsg,
+        ),
+      );
+    }
+  }
+
+  Future<void> resetToDefault() async {
+    emit(
+      state.copyWith(
+        status: ShortcutsStatus.updating,
+        error: '',
+      ),
+    );
+    try {
+      await service.saveAllShortcuts(defaultCommandShortcutEvents);
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.success,
+          commandShortcutEvents: defaultCommandShortcutEvents,
+          error: '',
+        ),
+      );
+    } catch (e) {
+      emit(
+        state.copyWith(
+          status: ShortcutsStatus.failure,
+          error: kCouldNotSaveErrorMsg,
+        ),
+      );
+    }
+  }
+
+  ///Checks if the new command is conflicting with other shortcut
+  ///We also check using the key, whether this command is a codeblock
+  ///shortcut, if so we only check a conflict with other codeblock shortcut.
+  String getConflict(CommandShortcutEvent currentShortcut, String command) {
```

Comment:
Is there a reason this returns a String and not eg. a boolean?

Reply:
Yes because we want to get the shortcut which is already using a keyBinding. You see the flow of execution here is as follows:
1. The user tries to update a shortcut with new key binding.
2. We check if the new key binding is already used by any existing shortcut.
3. If yes, we display the name of the shortcut for which the new key binding is already used.

 This is helpful because then the user gets to know what is causing a conflict, instead of only knowing that there is a conflict.